catalog/solr/catalog-solr-provider/src/main/java/ddf/catalog/solr/provider/SolrCatalogProvider.java

```diff
@@ -30,4 +32,13 @@ public SolrCatalogProvider(
         solrFilterDelegateFactory,
         null);
   }
+
+  public void setAnyTextAttributes(List<String> anyTextAttributes) {
+    super.resolver = null;
+    this.anyTextAttributes.clear();
+    if (CollectionUtils.isNotEmpty(anyTextAttributes)) {
```

Comment:
❗️ Do we support the list to be `null`? I don't think we should and as such, we should only need to call its `.isEmpty()` method. In any case, you would get a NPE in the above line. In addition. `addAll()` would take care of the case when there is no elements in the list so no need to check.
❗️ I am still concern about multi-threading issue since we first clear the list before adding to it.